In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import datetime as dt


In [2]:
customer=pd.read_csv("CUSTOMERS.CSV")
geo_location=pd.read_csv("GEO_LOCATION.csv")
order_items = pd.read_csv("ORDER_ITEMS.csv")
order_payment=pd.read_csv("ORDER_PAYMENTS.csv")
order_review=pd.read_csv("ORDER_REVIEW_RATINGS.csv")
orders=pd.read_csv("ORDERS.csv")
products=pd.read_csv("PRODUCTS.csv")
sellers=pd.read_csv("SELLERS.csv")

In [3]:
customer.rename(columns={"customer_zip_code_prefix": "zip_code"},inplace=True)
customer.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   customer_id         99441 non-null  object
 1   customer_unique_id  99441 non-null  object
 2   zip_code            99441 non-null  int64 
 3   customer_city       99441 non-null  object
 4   customer_state      99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [4]:
geo_location.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19015 entries, 0 to 19014
Data columns (total 5 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   geolocation_zip_code_prefix  19015 non-null  int64  
 1   geolocation_lat              19015 non-null  float64
 2   geolocation_lng              19015 non-null  float64
 3   geolocation_city             19015 non-null  object 
 4   geolocation_state            19015 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 742.9+ KB


In [5]:
geo_location.rename(columns={"geolocation_zip_code_prefix":"zip_code"},inplace=True)


In [6]:
df=pd.merge(left=customer,right=geo_location,on="zip_code",how="inner")

In [7]:
df1=pd.merge(left=order_items,right=order_payment,on="order_id",how="inner")
df1=pd.merge(left=df1,right=order_review,on="order_id",how="inner")
df1=pd.merge(left=df1,right=orders,on="order_id",how="inner")
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118315 entries, 0 to 118314
Data columns (total 22 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   order_id                       118315 non-null  object 
 1   order_item_id                  118315 non-null  int64  
 2   product_id                     118315 non-null  object 
 3   seller_id                      118315 non-null  object 
 4   shipping_limit_date            118315 non-null  object 
 5   price                          118315 non-null  float64
 6   freight_value                  118315 non-null  float64
 7   payment_sequential             118315 non-null  int64  
 8   payment_type                   118315 non-null  object 
 9   payment_installments           118315 non-null  int64  
 10  payment_value                  118315 non-null  float64
 11  review_id                      118315 non-null  object 
 12  review_score                  

In [8]:
df1.isnull().sum()

order_id                            0
order_item_id                       0
product_id                          0
seller_id                           0
shipping_limit_date                 0
price                               0
freight_value                       0
payment_sequential                  0
payment_type                        0
payment_installments                0
payment_value                       0
review_id                           0
review_score                        0
review_creation_date                0
review_answer_timestamp             0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                  15
order_delivered_carrier_date     1254
order_delivered_customer_date    2588
order_estimated_delivery_date       0
dtype: int64

### null values
order approved at = 15 
order_delivered_carrier_date = 1254
order_delivered_customer_date = 2588


In [9]:
(df1.isnull().sum()*100)/len(df1)

order_id                         0.000000
order_item_id                    0.000000
product_id                       0.000000
seller_id                        0.000000
shipping_limit_date              0.000000
price                            0.000000
freight_value                    0.000000
payment_sequential               0.000000
payment_type                     0.000000
payment_installments             0.000000
payment_value                    0.000000
review_id                        0.000000
review_score                     0.000000
review_creation_date             0.000000
review_answer_timestamp          0.000000
customer_id                      0.000000
order_status                     0.000000
order_purchase_timestamp         0.000000
order_approved_at                0.012678
order_delivered_carrier_date     1.059883
order_delivered_customer_date    2.187381
order_estimated_delivery_date    0.000000
dtype: float64

In [16]:
df1 = df1.dropna()

In [17]:
df1.isnull().sum()

order_id                         0
order_item_id                    0
product_id                       0
seller_id                        0
shipping_limit_date              0
price                            0
freight_value                    0
payment_sequential               0
payment_type                     0
payment_installments             0
payment_value                    0
review_id                        0
review_score                     0
review_creation_date             0
review_answer_timestamp          0
customer_id                      0
order_status                     0
order_purchase_timestamp         0
order_approved_at                0
order_delivered_carrier_date     0
order_delivered_customer_date    0
order_estimated_delivery_date    0
dtype: int64

In [18]:
df.isnull().sum()

customer_id           0
customer_unique_id    0
zip_code              0
customer_city         0
customer_state        0
geolocation_lat       0
geolocation_lng       0
geolocation_city      0
geolocation_state     0
dtype: int64